### PR TITLE
CR-18810-fix-create-gs

### PR DESCRIPTION
**Pull request by ghost**
## What
* fixed `cf git-source create` command to not crash (and handle runtime namespace correctly)
* fixed `cf runtime uninstall` to get runtime namespace from platform

## Why
the `cf git-source create` command was crashing, the `cf runtime uninstall` was working (even when runtimeName !== runtimeNamespace), but in a less safe way

### CODE REVIEW

File: cmd/commands/cluster.go [hunk 1]
```diff
@@ -28,11 +28,11 @@ import (
 	"github.com/codefresh-io/cli-v2/pkg/util"
 	kubeutil "github.com/codefresh-io/cli-v2/pkg/util/kube"
 	kustutil "github.com/codefresh-io/cli-v2/pkg/util/kust"
-	platmodel "github.com/codefresh-io/go-sdk/pkg/codefresh/model"
-	"github.com/ghodss/yaml"
 
 	"github.com/Masterminds/semver/v3"
-	"github.com/argoproj-labs/argocd-autopilot/pkg/kube"
+	apkube "github.com/argoproj-labs/argocd-autopilot/pkg/kube"
```
Comment (ghost) [inline cmd/commands/cluster.go:33]: reorder and add alias

File: cmd/commands/common.go [hunk 2]
```diff
@@ -38,8 +38,7 @@ import (
 	"github.com/codefresh-io/cli-v2/pkg/util/kube"
 	routingutil "github.com/codefresh-io/cli-v2/pkg/util/routing"
 
-	"github.com/argoproj-labs/argocd-autopilot/pkg/fs"
-	"github.com/argoproj-labs/argocd-autopilot/pkg/git"
+	apfs "github.com/argoproj-labs/argocd-autopilot/pkg/fs"
```
Comment (ghost) [inline cmd/commands/common.go:41]: reorder and add alias

File: cmd/commands/common.go [hunk 3]
```diff
@@ -744,24 +742,6 @@ func suggestIscRepo(ctx context.Context, suggestedSharedConfigRepo string) (stri
...
-}
-
-func getRuntimeInstallationType(ctx context.Context, runtimeName string) (*platmodel.InstallationType, error) {
-	rt, err := getRuntime(ctx, runtimeName)
-	if err != nil {
-		return nil, err
-	}
-
-	return &rt.InstallationType, nil
-}
```
Comment (ghost) [inline cmd/commands/common.go:763]: no need for those two funcs, were used in install prerun func before

File: cmd/commands/git-source.go [hunk 4]
```diff
@@ -30,9 +30,9 @@ import (
 	routingutil "github.com/codefresh-io/cli-v2/pkg/util/routing"
 	wfutil "github.com/codefresh-io/cli-v2/pkg/util/workflow"
 
-	"github.com/argoproj-labs/argocd-autopilot/pkg/application"
-	"github.com/argoproj-labs/argocd-autopilot/pkg/fs"
-	"github.com/argoproj-labs/argocd-autopilot/pkg/git"
+	apapp "github.com/argoproj-labs/argocd-autopilot/pkg/application"
+	apfs "github.com/argoproj-labs/argocd-autopilot/pkg/fs"
+	apgit "github.com/argoproj-labs/argocd-autopilot/pkg/git"
```
Comment (ghost) [inline cmd/commands/git-source.go:35]: reorder and add alias

File: cmd/commands/git-source.go [hunk 5]
```diff
@@ -188,17 +188,22 @@ func NewGitSourceCreateCommand() *cobra.Command {
...
-				runtimeNamespace = namespace
+			runtimeName := args[0]
+			runtime, err := getRuntime(ctx, runtimeName)
+			if err != nil {
+				return err
+			}
+
+			runtimeNamespace := runtimeName
+			if runtime.Metadata.Namespace != nil {
+				runtimeNamespace = *runtime.Metadata.Namespace
```
Comment (ghost) [inline cmd/commands/git-source.go:199]: fix for `git-source create` command

File: cmd/commands/helm_test.go [hunk 6]
```diff
@@ -20,9 +20,10 @@ import (
 	"net/http"
 	"testing"
 
-	kubemocks "github.com/argoproj-labs/argocd-autopilot/pkg/kube/mocks"
 	cfgit "github.com/codefresh-io/cli-v2/pkg/git"
 	gitmocks "github.com/codefresh-io/cli-v2/pkg/git/mocks"
+
+	kubemocks "github.com/argoproj-labs/argocd-autopilot/pkg/kube/mocks"
```
Comment (ghost) [inline cmd/commands/helm_test.go:26]: reorder and add alias

File: cmd/commands/runtime.go [hunk 7]
```diff
@@ -27,21 +27,20 @@ import (
...
+	kubeutil "github.com/codefresh-io/cli-v2/pkg/util/kube"
+	routingutil "github.com/codefresh-io/cli-v2/pkg/util/routing"
 
 	"github.com/Masterminds/semver/v3"
 	apcmd "github.com/argoproj-labs/argocd-autopilot/cmd/commands"
-	"github.com/argoproj-labs/argocd-autopilot/pkg/fs"
+	apfs "github.com/argoproj-labs/argocd-autopilot/pkg/fs"
 	apgit "github.com/argoproj-labs/argocd-autopilot/pkg/git"
-	"github.com/argoproj-labs/argocd-autopilot/pkg/kube"
+	apkube "github.com/argoproj-labs/argocd-autopilot/pkg/kube"
```
Comment (ghost) [inline cmd/commands/runtime.go:43]: reorder and add alias

File: cmd/commands/runtime.go [hunk 8]
```diff
@@ -148,32 +147,35 @@ func runtimeUninstallCommandPreRunHandler(cmd *cobra.Command, args []string, opt
...
 	if err != nil {
 		return err
 	}
 
-	if *installationType == platmodel.InstallationTypeHelm {
+	if rt.InstallationType == platmodel.InstallationTypeHelm {
 		return errors.New("This runtime was installed using Helm, please use Helm to uninstall it as well.")
 	}
 
-	if !opts.SkipChecks {
```
Comment (ghost) [inline cmd/commands/runtime.go:160]: removed all usage of SkipChecks

File: cmd/commands/runtime.go [hunk 9]
```diff
@@ -148,32 +147,35 @@ func runtimeUninstallCommandPreRunHandler(cmd *cobra.Command, args []string, opt
...
+
+	opts.runtimeNamespace, _ = cmd.Flags().GetString("namespace")
+	if opts.runtimeNamespace == "" {
+		opts.runtimeNamespace = opts.RuntimeName
 	}
 
-	if !opts.Managed {
-		opts.kubeContext, err = getKubeContextName(cmd.Flag("context"), cmd.Flag("kubeconfig"))
+	if rt.Metadata.Namespace != nil {
+		opts.runtimeNamespace = *rt.Metadata.Namespace
```
Comment (ghost) [inline cmd/commands/runtime.go:170]: get namespace from platform runtime object

File: cmd/commands/runtime_install.go [hunk 10]
```diff
@@ -46,8 +46,8 @@ import (
 
 	"github.com/Masterminds/semver/v3"
 	apcmd "github.com/argoproj-labs/argocd-autopilot/cmd/commands"
-	"github.com/argoproj-labs/argocd-autopilot/pkg/application"
-	"github.com/argoproj-labs/argocd-autopilot/pkg/fs"
+	apapp "github.com/argoproj-labs/argocd-autopilot/pkg/application"
+	apfs "github.com/argoproj-labs/argocd-autopilot/pkg/fs"
```
Comment (ghost) [inline cmd/commands/runtime_install.go:50]: reorder and add alias

File: cmd/commands/runtime_install.go [hunk 11]
```diff
@@ -2191,3 +2192,22 @@ func patchClusterResourcesAppSet(fs fs.FS) error {
 	appSet.ObjectMeta.Labels[store.Get().LabelKeyCFInternal] = "true"
 	return fs.WriteYamls(name, appSet)
 }
+
+func getRuntimeNamespace(cmd *cobra.Command, runtimeName string, runtimeVersion *semver.Version) string {
```
Comment (ghost) [inline cmd/commands/runtime_install.go:2196]: moved from `runtime.go` to `runtime_install.go` - only used inside this file

File: docs/commands/cli-v2_runtime_uninstall.md [hunk 12]
```diff
@@ -40,7 +40,6 @@ cli-v2 runtime uninstall [RUNTIME_NAME] [flags]
   -n, --namespace string         If present, the namespace scope for this CLI request
       --repo string              Repository URL [GIT_REPO]
       --request-timeout string   The length of time to wait before giving up on a single server request. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests. (default "0")
-      --skip-checks              If true, will not verify that runtime exists before uninstalling
```
Comment (ghost) [inline docs/commands/cli-v2_runtime_uninstall.md:43]: marked as deprecated - so it doesn't appear in help text

**Review by danielm-codefresh — APPROVED** [review]

**Pull request merged**